### PR TITLE
Major Refactoring for Consistency

### PR DESCRIPTION
**Pull request by ybabts**
This pull request overhauls several aspects of the codebase to enhance consistency, readability, and extendability. These improvements fix numerous issues in the current version and refactor several features to improve clarity.

## Return Type Adjustments
```diff
- getName: (url: string) => Promise<string> | string;
+ getName: (url: string) => string;
```
The return types on some functions were very strange, like for example the `getName` function on each Registry has a return type of `Promise<string> | string`. There was currently no case that the function in it's implementations could return a `Promise` and be `async`. I removed all the unnecessary `async` and `Promise` returns to make it more clear what the return type will be for the functions.

## `nextVersion` Implementation Overhaul
The `nextVersion` function had cons insistent implementations. The way the next version was found varied between Registries and caused the result to be different for each one. One example is how the `deno.land/x` registry gets the next version by doing a loop, while the `cdn.jsdelivr.net` gets the latest version from the `.latest` property on the JSON response. __This lead to some of the `nextVersion` functions getting the next version, while others got the latest version.__

This also made a difference to what versions were considered for each `nextVersion` function. The `deno.land/std` and the `deno.land/x` registries ensure that the version is not a prerelease version, but the rest of the registries just get the latest version and don't verify that it's not a prerelease version. __This causes the version between registries to be different, even if they are the same module.__

The solution to this is to unify how versions are fetched by separating the version selection logic from the fetching logic. For this I made a function `fetchVersions` that just fetches all the versions and returns a `string[]` of them. This gives us a consistent interface to work with where we can sort and filter the versions as we please in any other function. So I also made a `fetchNextVersion` and `fetchLatestVersion` function. These functions don't need to be implemented by each Registry because they operated by calling `fetchVersions` which as a consistent interface.

## Renaming for Clarity
This pull request changes several names to make them more clear. Several functions with the name `get` at the start have been changed to `fetch` instead. The idea of switching from `get` to `fetch` is that it signifies that the function is fetching data from the web while `get` is getting the data from a resource. This also changes and combines `getNextVersionURL` and `getCurrentVersionURL` to `buildVersionedURL`.
```diff
- name
+ registryName
+ moduleName
- prefix
+ urlPrefix
- getName()
+ getModuleNameFromURL()
- getCurrentVersion()
+ getVersionFromURL()
- getNextVersion()
+ fetchVersions()
+ fetchNextVersion()
+ fetchLatestVersion()
- getRepository()
+ fetchRepository()
- getCurrentVersionURL()
- getNextVersionURL()
+ buildVersionedURL
```
## `Registry` Class Refactor
The `Registry` class has been changed to allow for better code re-usability and less repeated syntax. The `Registry` class has been changed from a class that creates objects that resemble itself to a singleton abstract class. This allows you to implement functions that are shared by all Registries without having to add a new implementation for each Registry. This change leverages the consistent interface to allow every `Registry` class to have `fetchNextVersion`, `fetchCurrentVersion`, `buildVersionedURL`, and `buildFetchErrorMessage`.

## JSDelivr Split
JS Delivr had some funky logic inside of it to switch based on the URL. This added unnecessary parameters to all the registry functions for this one Registry. JSDelivr was actually just two different Registries in one, so __this pull request separates them out into two different ones to accommodate for their different logic.__

## Error Messaging Enhancements
The error messages in the project were very inconsistent. Sometimes they had no message while other times they had different message formatting. This pull request adds a function to `Registry` called `buildFetchErrorMessage` that takes in the `moduleName` and outputs a formatting error message to make consistent messaging between Registry's implementations.

### CODE REVIEW

File: registries.ts
Comment (boywithkeyboard) [inline registries.ts:1]: What's the reason for deleting this file?

**Review by boywithkeyboard — COMMENTED** [review]
I need further information on some changes.

File: registries.ts
Comment (ybabts) [reply registries.ts:1]: The JSDelivr registry was split into two different Registries, one for /gh and one for /npm.

Comment (boywithkeyboard) [reply registries.ts:1]: Oh you just renamed it to `./registries/mod.ts`

Comment (ybabts) [reply registries.ts:1]: yeah `./registries` was renamed to `./registries/mod.ts`

**Review by boywithkeyboard — APPROVED** [review]
lgtm

**Pull request merged**